test_requests.py

```diff
@@ -411,6 +416,21 @@ def test_DIGESTAUTH_QUOTES_QOP_VALUE(self):
         r = requests.get(url, auth=auth)
         assert '"auth"' in r.request.headers['Authorization']
 
+    def test_DIGESTAUTH_THREADED(self):
```

Comment:
I'd say if my test is failling reliabily (98%) on old unsafe implementation, and first version of this pr, it is good enough. Then, there is the much bigger project of testing thread safeness of the whole library, and splitting the suite into several files. That is another much longer project

m2c